### PR TITLE
Typed i18n

### LINKED ISSUE #110: Type check `t` function to catch missing keys
We have a bunch of missing keys in the setup UI, these should be caught by the type checker to prevent in the future.

### PR DESCRIPTION
**Pull request by wbobeirne**
Fixes #110

### What This Does

* Update dependencies for better handling of i18n types
  * Updates `typescript` to `5.1.6`
  * Updates `i18next` to `23.2.11`
  * Updates `react-i18next` to `13.0.2`
  * These updates were all necessary to get a properly typed `t` function
* Add type definitions for `i18next` that type the `t` function using our language JSONs
  * The TS compiler will now throw an error if a key is misspelled, or not present in all languages
* Rename all keys to use `-` as a separator instead of `_`
  * See https://www.i18next.com/principles/fallback#language-fallback for more details, `_` is used for variants
* Fix misspelled keys
* Add missing keys in other languages

### CODE REVIEW

**Review by EthnTuttle — COMMENTED** [review]
Just a few things before I pull locally to test.

**Comment by EthnTuttle** [comment]
Is there an IDE setting that needs to be enabled to get the type checking to work? It's not working in my vs code setup.
I also tried adding `"strict": true` to the tsconfig as mentioned [here](https://www.i18next.com/overview/typescript), but that didn't change anything.

Screenshot 2023-07-22 at 7 37 48 AM
Screenshot 2023-07-22 at 7 38 11 AM
Screenshot 2023-07-22 at 7 38 31 AM

**Comment by wbobeirne** [comment]
@EthnTuttle make sure your IDE is using the projects typescript version, not the IDEs. If you're in vscode, hit cmd+shift+p and type "typescript version" and select the 5.1.6 in node_modules. Uncertain if other IDEs have that issue as well.

File: apps/guardian-ui/src/typings/i18next.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import 'i18next';
+
+// import translation from './languages/en.json';
+import { languages } from './languages';
+
+declare module 'i18next' {
+  interface CustomTypeOptions {
+    // resources: { translation: typeof translation };
+    resources: (typeof languages)[0];
```
[the remark this answers is not in the document]
Comment (wbobeirne) [reply apps/guardian-ui/src/typings/i18next.d.ts:9]: In my experience it loosens the type to any of them in the array, but I could probably make it more explicitly any item in the array.

**Comment by okjodom** [comment]
1. Reviewed, and tested locally. Looks good.
2. Rebased to resolve conflict after #108 but also because I saw an [opportunity to rework the footer translation](https://github.com/fedimint/ui/pull/108/files#r1271355767)
3. Tested again, looks good...

**Review by okjodom — APPROVED** [review]

**Pull request merged**